pkgs/shells/fish/wrapper.nix

```diff
@@ -1,52 +1,109 @@
-{ lib, writeShellApplication, fish, writeTextFile }:
-
-lib.makeOverridable ({
-  completionDirs ? [],
-  functionDirs ? [],
-  confDirs ? [],
-  pluginPkgs ? [],
-  localConfig ? "",
-  shellAliases ? {},
-  runtimeInputs ? []
+{
+  lib,
+  writeShellApplication,
+  fish,
+  writeTextFile,
 }:
 
-let
-  aliasesStr = builtins.concatStringsSep "\n"
-      (lib.mapAttrsToList (k: v: "alias ${k} ${lib.escapeShellArg v}") shellAliases);
+lib.makeOverridable (
+  /**
+    Create a wrapped fish shell binary with some plugins, completions,
+    configuration snippets and functions sourced from the given paths.
+    This provides a convenient way to test Fish plugins and scripts
+    without having to alter the environment.
 
-  shellAliasesFishConfig = writeTextFile {
-    name = "wrapfish.aliases.fish";
-    destination = "/share/fish/vendor_conf.d/aliases.fish";
+    # Example
+
+    ```nix
+    wrapFish {
+      pluginPkgs = with fishPlugins; [
+        pure
+        foreign-env
+      ];
+      completionDirs = [ ];
+      functionDirs = [ ];
+      confDirs = [ "/path/to/some/fish/init/dir/" ];
+      shellAliases = {
+        hello = "echo 'Hello World!'";
+        bye = "echo 'Bye World!'; exit";
+      };
+      runtimeInputs = with pkgs; [
+        curl
+        w3m
+      ];
+    }
+    ```
+
+    # Type
+
+    ```pseudocode
+    wrapFish :: {
+      completionDirs :: [Path] ? [ ],
```

Comment:
I would remove the default value from the Type. But leave the '?' to indicate that the presence of the Argument is optional.